### ISSUE TITLE
Open Door API

### ISSUE THREAD

**Issue by alexhalbi** [uhppoted/uhppoted-rest#2]
Thanks for the great project you are making here.
How can I add an endpoint to be able to open a door with the REST API via a simple HTTP GET command?

What would you suggest to do to add authentication to the API before being able to access it?

 Thanks!

**Comment by uhppoted** [comment]
Hi,

The 'open door' API slipped through the cracks because the original system was 'card only' - it was only added to the CLI and MQTT later and on request. I'll add it to the REST API now though - give me a day or two? 

re. authentication. The REST server is intended to be accessed via either a known gateway or from a verified client application (i.e. not from the wider Internet). For security it supports TLS and (optionally) client certificate validation i.e. only clients with a certificate issued by the CA that issued the server certificate (or a certificate in the client certificate store) will be able to connect.

This [post](https://blog.cloudflare.com/introducing-tls-client-auth) by CloudFlare probably explains it better than I do :-).

If you can describe your architecture I'll try be more specific about the steps.

**Comment by alexhalbi** [comment]
Hi,
Thank you very much for the quick answer. A day or two and the fact that you will add that is more than generous from you. 
Please do it with a GET request to easily get it to work with my system. I would like to be able to call it from another IoT device which then operates the relais to open the doors with its app. I will just create an encapsulated network for the two devices to interact with the REST API, that should then work fine. Maybe I will add a proxy docker in between for easy authentication.
Thanks again for the help!

**Comment by uhppoted** [comment]
Hi,

I've pushed an initial implementation to the [uhppoted-rest master branch](https://github.com/uhppoted/uhppoted-api/commit/a8404a5bffb427f999fe5d3f88af1635f64eb9f0). I'm afraid it's a `POST` request rather than a `GET` because that makes more sense in the context of a REST API. 

It is documented in the (updated) OpenAPI specification, but basically a sample request looks something like:
```
http://127.0.0.1:8080/uhppote/device/405419896/door/4
{
  "user-id": 192837465,
  "otp": 7782
}
```
The `user-id` should be the card number of a card that has permission to open the door (it checks :-)). The OTP is optional (and not yet implemented). 

Please be aware that this is still very much a work in progress though, so some caveats:

1. It's very much an early sketch of what I think the API should look like so it may yet change

2. It depends on the master branches of [uhppote-core](https://github.com/uhppoted/uhppote-core) and [uhppoted-api](https://github.com/uhppoted/uhppoted-api) so you'll have to update the `go.mod` if you're not building the whole `uhppoted` project.

3. Using an OTP for authorization is provisional and still needs to be implemented.

Let me know what you think ..

**Comment by uhppoted** [comment]
Ok, that looks like it's a wrap - the `open-door` API is implemented and the currently supported authorization mechanisms are documented in the [README](https://github.com/uhppoted/uhppoted-rest/blob/master/README.md#api-authorization).

I'm going to leave this issue open pending release of v0.6.4 (in a week or two).

**Comment by uhppoted** [comment]
Released in v0.6.4
